app/src/main/java/com/github/gotify/messages/provider/MessageFacade.java

```diff
@@ -52,9 +60,24 @@ public int getLastReceivedMessage() {
         return state.getLastReceivedMessage();
     }
 
-    public void delete(Message message) {
-        this.requester.asyncRemoveMessage(message);
+    public void deleteLocal(Message message) {
```

Comment:
While this method is most likely only executed in the ui thread, could we still make it synchronized, just to ensure that there are no race conditions later.

Reply:
I think other methods need to be synchronous as well in that case.